server/v2/types.go

```diff
@@ -20,3 +23,36 @@ type AppI[T transaction.Tx] interface {
 	GetGPRCMethodsToMessageMap() map[string]func() gogoproto.Message
 	GetStore() any
 }
+
+// ExportedApp represents an exported app state, along with
+// validators, consensus params and latest app height.
+type ExportedApp struct {
+	// AppState is the application state as JSON.
+	AppState json.RawMessage
+	// Height is the app's latest block height.
+	Height int64
+}
+
+// AppExporter is a function that dumps all app state to
```

Comment:
I would put the app exported in x/gentuil, we should not add the cosmos-db dep in server/v2.
Additionally we can clean-up some args.